test/Microsoft.AspNetCore.Authentication.WsFederation.Test/WsFederationTest.cs

```diff
@@ -51,36 +52,44 @@ public class WsFederationTest
             var response = await httpClient.GetAsync("/");
             var queryItems = QueryHelpers.ParseQuery(response.Headers.Location.Query);
 
-            // Send an invalid token and verify that the token is not honored
+            var request = new HttpRequestMessage(HttpMethod.Post, queryItems["wreply"]);
+            CopyCookies(response, request);
             var kvps = new List<KeyValuePair<string, string>>();
             kvps.Add(new KeyValuePair<string, string>("wa", "wsignin1.0"));
             kvps.Add(new KeyValuePair<string, string>("wresult", File.ReadAllText(@"ValidToken.xml")));
             kvps.Add(new KeyValuePair<string, string>("wctx", queryItems["wctx"]));
-            response = await httpClient.PostAsync(queryItems["wreply"], new FormUrlEncodedContent(kvps));
+            request.Content = new FormUrlEncodedContent(kvps);
+            response = await httpClient.SendAsync(request);
 
             Assert.Equal(HttpStatusCode.Found, response.StatusCode);
 
-            var request = new HttpRequestMessage(HttpMethod.Get, response.Headers.Location);
-            var cookies = SetCookieHeaderValue.ParseList(response.Headers.GetValues(HeaderNames.SetCookie).ToList());
-            foreach (var cookie in cookies)
-            {
-                if (cookie.Value.HasValue)
-                {
-                    request.Headers.Add(HeaderNames.Cookie, new CookieHeaderValue(cookie.Name, cookie.Value).ToString());
-                }
-            }
+            request = new HttpRequestMessage(HttpMethod.Get, response.Headers.Location);
+            CopyCookies(response, request);
             response = await httpClient.SendAsync(request);
 
             // Did the request end in the actual resource requested for
             Assert.Equal(WsFederationDefaults.AuthenticationScheme, await response.Content.ReadAsStringAsync());
         }
 
         [Fact]
-        public async Task ValidUnsolicitedTokenIsAccepted()
+        public async Task ValidUnsolicitedTokenIsRefused()
         {
             var httpClient = CreateClient();
 
-            // Send an invalid token and verify that the token is not honored
+            var kvps = new List<KeyValuePair<string, string>>();
```

Comment:
Nit: maybe its worth moving a lot of the common kvps init into a BuildFormContent() helper